packages/chat/components/feedback/src/feedback-container.template.ts

```diff
@@ -0,0 +1,323 @@
+   *
+   * @param {object} event Event object of the corrected value from input box
+   * @param {object} event.target input element
+   * @private
+   */
+  handleFeedback(event) {
+    let feedback = event.target.labelText;
+    if (!this._feedbacks.includes(feedback)) {
+      this._feedbacks.push(feedback);
+    } else {
+      this._feedbacks = this._feedbacks.filter((item) => item != feedback);
+    }
+    this.formData.feedback = this._feedbacks.join(' | ');
+  }
+
+  /**
+   * Method for rendering Feedback Modal
+   */
+  openModal() {
+    this.selection = null;
+    return html`
+      <cds-modal
+        id="feedback-modal"
+        ?open=${this.isModelOpen}
+        @cds-modal-closed=${() => (this.isModelOpen = false)}>
+        <cds-modal-header>
+          <cds-modal-close-button></cds-modal-close-button>
+          <cds-modal-heading>Provide Feedback</cds-modal-heading>
+        </cds-modal-header>
+        <cds-modal-body>
+          <cds-form-group>
+            <cds-text-input
+              label="Selected Text:"
+              value=${this.formData.selected_value}
+              @input=${this.handleTextInput}>
+            </cds-text-input>
+          </cds-form-group>
+
+          <cds-form-group @cds-checkbox-changed=${this.handleFeedback}>
+            <cds-checkbox label-text="No problematic text found"></cds-checkbox>
+            <cds-checkbox
+              label-text="Contains HAP (e.g. hate, abusive language, profanity)"></cds-checkbox>
+            <cds-checkbox
+              label-text="Contains PII (e.g. SSN, VIN, personal address)"></cds-checkbox>
+            <cds-checkbox
+              label-text="Contains social bias (e.g. race, religion, social status, etc.)"></cds-checkbox>
+            <cds-checkbox label-text="Isn’t truthful/honest"></cds-checkbox>
+            <cds-checkbox
+              label-text="Contains Taboo Topics (eg. religion, politics etc.)"></cds-checkbox>
+            <cds-checkbox
+              label-text="Other problem (please provide detail)"></cds-checkbox>
+          </cds-form-group>
+
+          <cds-form-group>
+            <cds-textarea
+              rows="4"
+              style="padding:8px"
+              value=${this.formData.comment}
+              @input=${this.handleTextArea}>
+              <span slot="label-text">Comments:</span>
+            </cds-textarea>
+          </cds-form-group>
+        </cds-modal-body>
+        <cds-modal-footer>
+          <cds-modal-footer-button kind="secondary" data-modal-close
+            >Cancel</cds-modal-footer-button
+          >
+          <cds-modal-footer-button kind="primary" @click=${this._handleFormData}
+            >Save</cds-modal-footer-button
+          >
+        </cds-modal-footer>
+      </cds-modal>
+    `;
+  }
+
+  /**
+   * Method for toggling the Feedback Modal
+   */
+  _toggle() {
+    this.isModelOpen = !this.isModelOpen;
+  }
+
+  handleSlotChange() {}
+
+  render() {
```

Comment:
We want to break the component out to separate parts so that adopters who want to extend the component can take in certain parts and adjust as needed.

If you take a look at the `test-input` component, we have `packages/chat/components/test-input/src/test-input.ts` file that contains all the class properties and method. We then import the class into the `packages/chat/components/test-input/src/test-input.template.ts` file which contains what you would put in the `render()`. And finally we have the `packages/chat/components/test-input/test-input.ts` file that take both the class file (`src/test-input.ts`) and template file (`src/test-input.template.ts`) and registers the component with the `@customElement()` decorator.

So for this `feedback-container`, let's break this `feedback-container.template.ts` file out 

`src/feedback-container.ts` -> class methods and properties
`src/feedback-container.template.ts` -> the render()
`feedback-container.ts` -> bring in both files - view `test-input.ts` for reference